charmcraft/commands/build.py

```diff
@@ -178,12 +178,29 @@ def run(self) -> List[str]:
         """
         charms: List[str] = []
 
+        # Validate base indices, if any.
+        if bases_indices:
+            invalid_indices = [
+                i for i in bases_indices if i < 0 or i >= len(self.config.bases)
+            ]
+            if invalid_indices:
+                raise CommandError(
+                    f"Specified base index '{invalid_indices[0]}' is out of range."
```

Comment:
If one element will be just shown, I'd do a simple "for and raise":
```
for i in bases_indices:
    if i < 0 or i >= len(self.config.bases):
        raise CommandError(...)
```

Reply:
Done (and moved it its own method)!